src/main/java/com/zaxxer/hikari/HikariConnectionCreator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2022 Brett Wooldridge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.zaxxer.hikari;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import javax.sql.DataSource;
+
+/**
```

Comment:
Tell users that the param `datasource` is the underlying DataSource instance(implemented by JDBC Driver), which created by HikariDataSource. 
Because HikariDataSource#getConnection(Strinig, String) is not implemented:
```java
   @Override
   public Connection getConnection(String username, String password) throws SQLException
   {
      throw new SQLFeatureNotSupportedException();
   }
```